homeassistant/components/elmax/__init__.py

```diff
@@ -0,0 +1,62 @@
+"""The elmax-cloud integration."""
+from __future__ import annotations
+
+from datetime import timedelta
+import logging
+
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.helpers.typing import HomeAssistantType
+
+from .common import ElmaxCoordinator
+from .const import (
+    CONF_ELMAX_PANEL_ID,
+    CONF_ELMAX_PANEL_PIN,
+    CONF_ELMAX_PASSWORD,
+    CONF_ELMAX_USERNAME,
+    DOMAIN,
+    ELMAX_PLATFORMS,
+    POLLING_SECONDS,
+)
+
+_LOGGER = logging.getLogger(__name__)
+
+
+async def async_setup_entry(hass: HomeAssistantType, entry: ConfigEntry) -> bool:
+    """Set up elmax-cloud from a config entry."""
+    hass.data.setdefault(DOMAIN, {})
```

Comment:
```suggestion
```

Reply:
We should not set anything on `hass.data` before the entry has not been successfully set up

Reply:
Agreed.